tests/baselines/reference/indirectDiscriminantAndExcessProperty.errors.txt

```diff
@@ -14,7 +14,7 @@ indirectDiscriminantAndExcessProperty.ts(22,5): error TS2322: Type 'string' is n
     thing({
         type: foo1,
         ~~~~
-!!! error TS2322: Type 'string' is not assignable to type '"foo" | "bar"'.
+!!! error TS2322: Type 'string' is not assignable to type '"foo"'.

```

Comment:
Heh - this is kind of better if you're already familiar with string literals, but I'd argue overall worse in that it doesn't give as much of a hint that `type` is really the discriminant.

Reply:
Note that this isn't related to discriminants here. This case just happens to have a discriminant target.

The reason this has changed is that previously `getBestMatchIndexedAccessTypeOrUndefined` would prefer the simple match done using `getIndexedAccessTypeOrUndefined`. Since `type` is a shared property in the union target that would return early and the error with the union type would be raised. Now `getBestMatchingType` is preferred so this new error here is the product of the best overlappy type matching.

This now matches more closely the non-elementwise error reported in the very same situation, see the playground [here](https://www.typescriptlang.org/play?ts=5.5.0-dev.20240226#code/KYDwDg9gTgLgBDAnmYcBCAbAhgCzgXgCg4S4AfOAbwWWAC44AiAMwgkYBo4sAjAYwYBnGFACWAOwDmcAL7FSFakhQNGPLFE5wQiAF4NxAVwC2PYFC6gRWBlnGJZAbkKEAJsD7YoqZofF8YUQhxBBwJSQAKHmwcBkxcAEoGADcIUVdnQgxgeFYIAEYCJjzGZxgwqQjKeRJlejg8-I4a7n5VHGAMDAgAQkZCGQTMvmDhOAgeACsiqha6hkbm0laBJg6u3v65cvCIicmhwiA)